aws/session/credentials.go

```diff
@@ -206,7 +207,14 @@ func credsFromAssumeRole(cfg aws.Config,
 		sharedCfg.RoleARN,
 		func(opt *stscreds.AssumeRoleProvider) {
 			opt.RoleSessionName = sharedCfg.RoleSessionName
-			opt.Duration = sessOpts.AssumeRoleDuration
+
+			if sessOpts.AssumeRoleDuration == 0 &&
+				sharedCfg.RoleDurationSeconds != nil &&
+				*sharedCfg.RoleDurationSeconds/time.Minute > 15 {
+				opt.Duration = *sharedCfg.RoleDurationSeconds
+			} else if sessOpts.AssumeRoleDuration != 0 {
+				opt.Duration = sessOpts.AssumeRoleDuration
+			}
```

Comment:
What do you think about `sharedConfig.RoleDurationSeconds` and the `session.Options.AssumeRoleDuration` sharing the same name? ,e.g AssumeRoleDuration.  This may also help any confusion caused by having `Seconds` in the `RoleDurationSeconds` member name. Even tthough the type is a time.Duration.